jujugui/static/gui/src/test/test_ghost_deployer_extension.js

```diff
@@ -145,6 +145,15 @@ describe('Ghost Deployer Extension', function() {
     assert.equal(ghostDeployer.env.deploy.callCount, 1);
   });
 
+  it('does not call addCharm for local charms and juju 2', function() {
```

Comment:
Please add two similar tests, one for Juju 1 and one for Juju 2. If you agree with my comment above AddCharm should not be called in both cases.